arch/x86/Makefile.uk

```diff
@@ -5,7 +5,13 @@ endif
 
 $(eval $(call addlib,libcontext))
 
+# TODO: Remove this dependency in the context of plat re-arch
```

Comment:
In the same spirit as in https://github.com/unikraft/unikraft/pull/1322#discussion_r1532162561, this code should not be in `arch`. Even more with the call to `UK_CTOR_PRIO`, as nothing in `arch` should depend on definitions in `lib`, `drivers`, or `include`. I would propose that we keep that code under `plat/` and later make it part of `libuklcpu`.

Also the commit message mentions `libcontext`, perhaps it needs to be updated.

Reply:
**Update:** During an offline discussion with @skuenzer it was agreed that we merge this now, as migration involves additional code that existed previously in `arch`, and do a migration once all depending PRs are merged.